pkg/policy/distillery.go

```diff
@@ -115,25 +100,24 @@ func (cache *PolicyCache) updateSelectorPolicy(identity *identityPkg.Identity) (
 	defer cip.Unlock()
 
 	// Don't resolve policy if it was already done for this or later revision.
-	if cip.getPolicy().Revision >= cache.repo.GetRevision() {
-		return false, nil
+	if policy := cip.getPolicy(); policy != nil && policy.Revision >= cache.repo.GetRevision() {
+		return cip, false, nil
 	}
 
 	// Resolve the policies, which could fail
 	selPolicy, err := cache.repo.resolvePolicyLocked(identity)
 	if err != nil {
-		return false, err
+		return nil, false, err
 	}
 
 	cip.setPolicy(selPolicy)
 
-	return true, nil
+	return cip, true, nil
 }
 
-// LocalEndpointIdentityAdded creates a SelectorPolicy cache entry for the
-// specified Identity, without calculating any policy for it.
+// LocalEndpointIdentityAdded is not needed; we only care about local endpoint
+// deletion
 func (cache *PolicyCache) LocalEndpointIdentityAdded(identity *identityPkg.Identity) {
-	cache.insert(identity)
 }
```

Comment:
Maybe delete the function then? Looks like this is the only implementation in production code, so maybe this function can be removed from `identitymanager.Observer` interface?

Reply:
I actually have a use for this in the very near future, so let's keep it for now.